source/ButtonModel.h

```diff
@@ -68,6 +70,7 @@ class ButtonModel {
 	 */
 	virtual bool init(const float a) {
 		this->angle = a;
+		isActive = false;
```

Comment:
Initializing a model by definition should activate it.

Reply:
these init methods are only used in the allocation of objects for the object pool, so it doesn't make sense for them to start as active. active in this case means placed.

Reply:
Init should be called when the object is placed. The constructor is used to allocate for the object pool. If we're using the init methods to add to the object pool, we have a bigger refactor on our hands.

Reply:
init is called in the alloc method in basically every class in the project

Reply:
Right, because the static alloc methods are used as a substitute for the normal constructor, in that we want to spawn an object immediately. If we're using an object pool, we're not supposed to be using those. Looks like we have some cleaning up to do.

Actually, aren't basically all our models stored in object pools? I'm not sure we even need to keep Walker's old alloc methods around.